plugin/pkg/admission/podpreset/admission.go

```diff
@@ -183,16 +185,11 @@ func safeToApplyPodPresetsOnPod(pod *api.Pod, podPresets []*settingsv1alpha1.Pod
 	if _, err := mergeVolumes(pod.Spec.Volumes, podPresets); err != nil {
 		errs = append(errs, err)
 	}
-	for _, ctr := range pod.Spec.Containers {
-		if err := safeToApplyPodPresetsOnContainer(&ctr, podPresets); err != nil {
+	pods.VisitContainers(&pod.Spec, func(c *api.Container, _ *field.Path) {
+		if err := safeToApplyPodPresetsOnContainer(c, podPresets); err != nil {
```

Comment:
Related to my earlier comment, how would a reviewer tell this case apart from the "get error out of the closure" case?  Structurally if the only difference is a single `:` I worry that it's not obvious.

Reply:
reflecting on it a bit further, an even more defensive method would be to use both:
```
var retErr error
podutil.VisitContainers(&pod.Spec, func(c *v1.Container) bool {
  if err := checkContainer(c); err != nil {
    retErr = err
    return false
  }
  return true
})
```
I don't think this is really necessary, though, so I didn't include it in my update.